File: Lab0/src/count_seq.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define N 100000000
#define TARGET 42

int main() {

    int *arr = malloc(N * sizeof(int));
    if (!arr) {
        perror("malloc");
        return 1;
    }

    srand(time(NULL));

    for (long i = 0; i < N; i++)
        arr[i] = rand() % 100;

    clock_t start = clock();

    long count = 0;
    for (long i = 0; i < N; i++)
        if (arr[i] == TARGET)
            count++;

    clock_t end = clock();

    double elapsed =
        (double)(end - start) / CLOCKS_PER_SEC;

    printf("Count = %ld\n", count);
    printf("Time = %.2f seconds\n", elapsed);

    free(arr);
    return 0;
}

File: Lab0/src/count_seq.py
import random
import time

# Number of elements
N = 100_000
TARGET = 42

# Create array
arr = [random.randint(0, 99) for _ in range(N)]

# Start timing
start = time.time()

# Count occurrences
count = sum(1 for x in arr if x == TARGET)

# End timing
end = time.time()

print(f"Count = {count}")
print(f"Time = {end - start:.2f} seconds")